Sources/Sentry/Profiling/SentryContinuousProfiler.mm

```diff
@@ -98,16 +104,30 @@ + (BOOL)isCurrentlyProfiling
 
 + (void)stop
 {
-    std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
+    {
+        std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
 
-    if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
-        SENTRY_LOG_DEBUG(@"No continuous profiler is currently running.");
-        return;
+        if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
+            SENTRY_LOG_DEBUG(@"No continuous profiler is currently running.");
+            return;
+        }
+
+        _sentry_threadUnsafe_transmitChunkEnvelope();
+        disableTimer();
     }
 
-    _sentry_threadUnsafe_transmitChunkEnvelope();
-    disableTimer();
-    [_threadUnsafe_gContinuousCurrentProfiler stopForReason:SentryProfilerTruncationReasonNormal];
+    {
+        std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
+        [_threadUnsafe_gContinuousCurrentProfiler
+            stopForReason:SentryProfilerTruncationReasonNormal];
+        _threadUnsafe_gContinuousCurrentProfiler = nil;
+    }
```

Comment:
l: Curious question: Why are you splitting this function into two scopes that use the same lock?

Reply:
I can't remember now why this happened. I might've been debugging a deadlock. I don't see anything bad happening if I combine the scopes, I think there's no issue there.